student-info-system/student/views.py

```diff
@@ -140,7 +140,7 @@ def history(request):
     data = {
         'auser': the_user,
     }
-    ssects = student_ssects_by_sem(student)
+    ssects = student.ssects_by_sem()
```

Comment:
I see why you wanted student_ssectes_by_sem, but I'm not sure that's worthy of a function -- instead of just doing the order_by on student.sectionstudent_set.

I'd put the gpa in a bespoke dict "sem_gpas" with { sem_id: gpa } entries and pull that out in the transcript when the semester id changes. Smoother than zipping.

Reply:
I'm not really sure what you're asking for. It seems like you have a specific way you want this, so I'll just leave it up to you and work on something else since I'm not really understanding what you're asking for.